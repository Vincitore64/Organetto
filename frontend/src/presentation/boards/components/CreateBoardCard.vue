<template>
  <a-card hoverable class="create-card"
    :body-style="{ height: 'max-content', display: 'flex', flexDirection: 'column', justifyContent: 'center', padding: '24px' }"
    @click="$emit('create')">
    <template #cover>
      <div class="create-cover">
        <div class="gradient-bg">
          <div class="floating-icons">
            <div class="floating-icon" v-for="n in 6" :key="n" :style="getFloatingIconStyle(n)">
              <component :is="getRandomIcon()" />
            </div>
          </div>
        </div>
      </div>
    </template>

    <div class="create-body">
      <div class="main-icon">
        <plus-outlined class="plus-icon" />
      </div>
      <h3 class="create-title">{{ t('boards.page.create') }}</h3>
      <p class="create-description">Создайте новую доску для вашего проекта</p>
      <div class="create-features">
        <div class="feature">
          <CheckOutlined class="feature-icon" />
          <span>Шаблоны</span>
        </div>
        <div class="feature">
          <CheckOutlined class="feature-icon" />
          <span>Совместная работа</span>
        </div>
      </div>
    </div>
  </a-card>
</template>

<script setup lang="ts">
import { useI18n } from 'vue-i18n'
import {
  PlusOutlined,
  CheckOutlined,
  BulbOutlined,
  RocketOutlined,
  StarOutlined,
  HeartOutlined,
  ThunderboltOutlined,
  CrownOutlined
} from '@ant-design/icons-vue'

const { t } = useI18n()

defineEmits<{
  create: []
}>()

const icons = [
  BulbOutlined,
  RocketOutlined,
  StarOutlined,
  HeartOutlined,
  ThunderboltOutlined,
  CrownOutlined
]

const getRandomIcon = () => {
  return icons[Math.floor(Math.random() * icons.length)]
}

const getFloatingIconStyle = (index: number) => {
  const positions = [
    { top: '10%', left: '15%', animationDelay: '0s' },
    { top: '20%', right: '20%', animationDelay: '0.5s' },
    { top: '40%', left: '10%', animationDelay: '1s' },
    { top: '60%', right: '15%', animationDelay: '1.5s' },
    { top: '75%', left: '25%', animationDelay: '2s' },
    { top: '30%', right: '35%', animationDelay: '2.5s' }
  ]
  return positions[index - 1] || {}
}
</script>

<style scoped lang="scss">
.create-card {
  border-radius: 12px;
  border: 2px dashed var(--color-primary-300, #91d5ff);
  background: var(--color-surface, #ffffff);
  transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
  cursor: pointer;
  overflow: hidden;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.06);
  min-height: 320px;

  &:hover {
    transform: translateY(-4px);
    box-shadow: 0 8px 32px rgba(0, 0, 0, 0.12);
    border-color: var(--color-primary-500, #1890ff);
    background: linear-gradient(135deg, #ffffff, #f6ffed);

    .main-icon {
      transform: scale(1.1) rotate(90deg);
    }

    .floating-icon {
      animation-play-state: running;
    }

    .gradient-bg {
      background: linear-gradient(135deg,
          rgba(24, 144, 255, 0.1),
          rgba(82, 196, 26, 0.1),
          rgba(250, 173, 20, 0.1));
    }
  }

  :deep(.ant-card-body) {
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 24px;
  }
}

.create-cover {
  height: 120px;
  position: relative;
  overflow: hidden;

  .gradient-bg {
    width: 100%;
    height: 100%;
    background: linear-gradient(135deg,
        rgba(240, 242, 245, 0.8),
        rgba(230, 247, 255, 0.8));
    transition: background 0.3s ease;
    position: relative;
  }

  .floating-icons {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;

    .floating-icon {
      position: absolute;
      font-size: 16px;
      color: var(--color-primary-400, #69c0ff);
      opacity: 0.6;
      animation: float 3s ease-in-out infinite;
      animation-play-state: paused;

      @keyframes float {

        0%,
        100% {
          transform: translateY(0px) rotate(0deg);
        }

        50% {
          transform: translateY(-10px) rotate(180deg);
        }
      }
    }
  }
}

.create-body {
  text-align: center;

  .main-icon {
    display: inline-flex;
    align-items: center;
    justify-content: center;
    width: 64px;
    height: 64px;
    border-radius: 50%;
    background: linear-gradient(135deg, var(--color-primary-500, #1890ff), var(--color-primary-600, #096dd9));
    margin-bottom: 16px;
    transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
    box-shadow: 0 4px 16px rgba(24, 144, 255, 0.3);

    .plus-icon {
      font-size: 28px;
      color: white;
    }
  }

  .create-title {
    font-size: 18px;
    font-weight: 600;
    color: var(--color-text, #262626);
    margin: 0 0 8px 0;
    line-height: 1.4;
  }

  .create-description {
    font-size: 14px;
    color: var(--color-text-weak, #8c8c8c);
    margin: 0 0 16px 0;
    line-height: 1.5;
  }

  .create-features {
    display: flex;
    justify-content: center;
    gap: 16px;

    .feature {
      display: flex;
      align-items: center;
      gap: 4px;
      font-size: 12px;
      color: var(--color-success, #52c41a);

      .feature-icon {
        font-size: 12px;
      }
    }
  }
}
</style>