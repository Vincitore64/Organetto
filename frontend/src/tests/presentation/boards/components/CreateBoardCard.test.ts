import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import CreateBoardCard from '@/presentation/boards/components/CreateBoardCard.vue'
import { createI18n } from 'vue-i18n'

// Mock i18n
const i18n = createI18n({
  legacy: false,
  locale: 'en',
  messages: {
    en: {
      boards: {
        page: {
          create: 'Create Board',
        },
      },
    },
    ru: {
      boards: {
        page: {
          create: 'Создать доску',
        },
      },
    },
  },
})

// Mock Ant Design Vue icons
const mockIcons = {
  PlusOutlined: { template: '<span class="plus-icon">+</span>' },
  CheckOutlined: { template: '<span class="check-icon">✓</span>' },
  BulbOutlined: { template: '<span class="bulb-icon">💡</span>' },
  RocketOutlined: { template: '<span class="rocket-icon">🚀</span>' },
  StarOutlined: { template: '<span class="star-icon">⭐</span>' },
  HeartOutlined: { template: '<span class="heart-icon">❤️</span>' },
  ThunderboltOutlined: { template: '<span class="thunder-icon">⚡</span>' },
  CrownOutlined: { template: '<span class="crown-icon">👑</span>' },
}

// Mock Ant Design Vue Card component
const mockACard = {
  template: `
    <div class="ant-card" :class="{ 'ant-card-hoverable': hoverable }">
      <div class="ant-card-cover" v-if="$slots.cover">
        <slot name="cover"></slot>
      </div>
      <div class="ant-card-body" :style="bodyStyle">
        <slot></slot>
      </div>
    </div>
  `,
  props: ['hoverable', 'bodyStyle'],
}

const createWrapper = (props = {}) => {
  return mount(CreateBoardCard, {
    props,
    global: {
      plugins: [i18n],
      components: {
        ...mockIcons,
        'a-card': mockACard,
      },
    },
  })
}

describe('CreateBoardCard.vue', () => {
  // 1. Rendering tests
  describe('Rendering', () => {
    it('renders the main structure correctly', () => {
      const wrapper = createWrapper()

      expect(wrapper.find('main').exists()).toBe(true)
      expect(wrapper.find('.create-card').exists()).toBe(true)
      expect(wrapper.find('.create-cover').exists()).toBe(true)
      expect(wrapper.find('.create-body').exists()).toBe(true)
    })

    it('renders the create title with i18n', () => {
      const wrapper = createWrapper()

      expect(wrapper.find('.create-title').text()).toBe('Create Board')
    })

    it('renders the create description', () => {
      const wrapper = createWrapper()

      expect(wrapper.find('.create-description').text()).toBe(
        'Создайте новую доску для вашего проекта',
      )
    })

    it('renders the main plus icon', () => {
      const wrapper = createWrapper()

      expect(wrapper.find('.main-icon').exists()).toBe(true)
      expect(wrapper.find('.plus-icon').exists()).toBe(true)
    })

    it('renders feature list with check icons', () => {
      const wrapper = createWrapper()

      const features = wrapper.findAll('.feature')
      expect(features).toHaveLength(2)

      expect(features[0].text()).toContain('Шаблоны')
      expect(features[1].text()).toContain('Совместная работа')

      expect(wrapper.findAll('.check-icon')).toHaveLength(2)
    })

    it('renders floating icons in the cover', () => {
      const wrapper = createWrapper()

      const floatingIcons = wrapper.findAll('.floating-icon')
      expect(floatingIcons).toHaveLength(6)
    })

    it('renders gradient background in cover', () => {
      const wrapper = createWrapper()

      expect(wrapper.find('.gradient-bg').exists()).toBe(true)
    })
  })

  // 2. Interaction tests
  describe('Interactions', () => {
    it('emits "create" event when card is clicked', async () => {
      const wrapper = createWrapper()

      await wrapper.find('.create-card').trigger('click')

      expect(wrapper.emitted()).toHaveProperty('create')
      expect(wrapper.emitted('create')).toHaveLength(1)
    })

    it('emits "create" event when main element is clicked', async () => {
      const wrapper = createWrapper()

      await wrapper.find('main').trigger('click')

      expect(wrapper.emitted()).toHaveProperty('create')
    })

    it('has hoverable card property set', () => {
      const wrapper = createWrapper()

      const card = wrapper.findComponent({ name: 'a-card' })
      expect(card.props('hoverable')).toBe(true)
    })
  })

  // 3. Component methods and computed properties
  describe('Component Logic', () => {
    it('getRandomIcon returns a valid icon component', () => {
      const wrapper = createWrapper()
      const vm = wrapper.vm as any

      // Test multiple calls to ensure randomness works
      for (let i = 0; i < 10; i++) {
        const icon = vm.getRandomIcon()
        expect(icon).toBeDefined()
        expect(typeof icon).toBe('object')
      }
    })

    it('getFloatingIconStyle returns correct positioning for each index', () => {
      const wrapper = createWrapper()
      const vm = wrapper.vm as any

      // Test all 6 positions
      for (let i = 1; i <= 6; i++) {
        const style = vm.getFloatingIconStyle(i)
        expect(style).toBeDefined()
        expect(style).toHaveProperty('animationDelay')
        expect(typeof style.animationDelay).toBe('string')
      }
    })

    it('getFloatingIconStyle returns empty object for invalid index', () => {
      const wrapper = createWrapper()
      const vm = wrapper.vm as any

      const style = vm.getFloatingIconStyle(0)
      expect(style).toEqual({})

      const style2 = vm.getFloatingIconStyle(7)
      expect(style2).toEqual({})
    })
  })

  // 4. CSS classes and styling
  describe('Styling', () => {
    it('has correct CSS classes applied', () => {
      const wrapper = createWrapper()

      expect(wrapper.find('.create-card').classes()).toContain('create-card')
      expect(wrapper.find('.create-cover').classes()).toContain('create-cover')
      expect(wrapper.find('.create-body').classes()).toContain('create-body')
      expect(wrapper.find('.main-icon').classes()).toContain('main-icon')
    })

    it('applies correct body style to card', () => {
      const wrapper = createWrapper()

      const card = wrapper.findComponent({ name: 'a-card' })
      const bodyStyle = card.props('bodyStyle')

      expect(bodyStyle).toEqual({
        height: 'max-content',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        padding: '16px 12px 0px 12px',
      })
    })

    it('floating icons have correct positioning styles', () => {
      const wrapper = createWrapper()
      const floatingIcons = wrapper.findAll('.floating-icon')

      // Check that each floating icon has inline styles applied
      floatingIcons.forEach((icon) => {
        const style = icon.attributes('style')
        expect(style).toBeDefined()
        expect(style).toContain('animation-delay')
      })
    })
  })

  // 5. Internationalization
  describe('Internationalization', () => {
    it('displays correct text in English locale', async () => {
      i18n.global.locale.value = 'en'
      const wrapper = createWrapper()

      await wrapper.vm.$nextTick()
      expect(wrapper.find('.create-title').text()).toBe('Create Board')
    })

    it('displays correct text in Russian locale', async () => {
      i18n.global.locale.value = 'ru'
      const wrapper = createWrapper()

      await wrapper.vm.$nextTick()
      expect(wrapper.find('.create-title').text()).toBe('Создать доску')
    })
  })

  // 6. Edge cases
  describe('Edge Cases', () => {
    it('handles missing i18n gracefully', () => {
      const wrapperWithoutI18n = mount(CreateBoardCard, {
        global: {
          components: {
            ...mockIcons,
            'a-card': mockACard,
          },
          mocks: {
            t: (key: string) => key,
          },
        },
      })

      expect(wrapperWithoutI18n.find('.create-title').text()).toBe('boards.page.create')
    })

    it('renders without crashing when no props are provided', () => {
      expect(() => createWrapper()).not.toThrow()
    })

    it('maintains component structure integrity', () => {
      const wrapper = createWrapper()

      // Ensure all critical elements exist
      expect(wrapper.find('main').exists()).toBe(true)
      expect(wrapper.find('.create-card').exists()).toBe(true)
      expect(wrapper.find('.main-icon').exists()).toBe(true)
      expect(wrapper.find('.create-title').exists()).toBe(true)
      expect(wrapper.find('.create-description').exists()).toBe(true)
      expect(wrapper.find('.create-features').exists()).toBe(true)
    })
  })

  // 7. Accessibility
  describe('Accessibility', () => {
    it('uses semantic HTML elements', () => {
      const wrapper = createWrapper()

      expect(wrapper.find('main').exists()).toBe(true)
      expect(wrapper.find('h3').exists()).toBe(true)
      expect(wrapper.find('p').exists()).toBe(true)
    })

    it('has clickable card for keyboard navigation', () => {
      const wrapper = createWrapper()

      const card = wrapper.find('.create-card')
      expect(card.exists()).toBe(true)
      // Card should be clickable for accessibility
    })
  })

  // 8. Snapshot test
  describe('Snapshot', () => {
    it('matches snapshot for create board card', () => {
      const wrapper = createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
