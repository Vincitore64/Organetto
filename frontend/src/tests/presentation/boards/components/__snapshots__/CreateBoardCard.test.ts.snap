// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreateBoardCard.vue > Snapshot > matches snapshot for create board card 1`] = `
"<main data-v-0f8075b3="">
  <div data-v-0f8075b3="" class="ant-card create-card">
    <div class="ant-card-cover">
      <div data-v-0f8075b3="" class="create-cover">
        <div data-v-0f8075b3="" class="gradient-bg">
          <div data-v-0f8075b3="" class="floating-icons">
            <div data-v-0f8075b3="" class="floating-icon" style="top: 10%; left: 15%; animation-delay: 0s;"><span data-v-0f8075b3="" role="img" aria-label="bulb" class="anticon anticon-bulb"><svg focusable="false" data-icon="bulb" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M632 888H392c-4.4 0-8 3.6-8 8v32c0 17.7 14.3 32 32 32h192c17.7 0 32-14.3 32-32v-32c0-4.4-3.6-8-8-8zM512 64c-181.1 0-328 146.9-328 328 0 121.4 66 227.4 164 284.1V792c0 17.7 14.3 32 32 32h264c17.7 0 32-14.3 32-32V676.1c98-56.7 164-162.7 164-284.1 0-181.1-146.9-328-328-328zm127.9 549.8L604 634.6V752H420V634.6l-35.9-20.8C305.4 568.3 256 484.5 256 392c0-141.4 114.6-256 256-256s256 114.6 256 256c0 92.5-49.4 176.3-128.1 221.8z"></path></svg><!----></span></div>
            <div data-v-0f8075b3="" class="floating-icon" style="top: 20%; right: 20%; animation-delay: 0.5s;"><span data-v-0f8075b3="" role="img" aria-label="star" class="anticon anticon-star"><svg focusable="false" data-icon="star" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"></path></svg><!----></span></div>
            <div data-v-0f8075b3="" class="floating-icon" style="top: 40%; left: 10%; animation-delay: 1s;"><span data-v-0f8075b3="" role="img" aria-label="star" class="anticon anticon-star"><svg focusable="false" data-icon="star" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"></path></svg><!----></span></div>
            <div data-v-0f8075b3="" class="floating-icon" style="top: 60%; right: 15%; animation-delay: 1.5s;"><span data-v-0f8075b3="" role="img" aria-label="crown" class="anticon anticon-crown"><svg focusable="false" data-icon="crown" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M899.6 276.5L705 396.4 518.4 147.5a8.06 8.06 0 00-12.9 0L319 396.4 124.3 276.5c-5.7-3.5-13.1 1.2-12.2 7.9L188.5 865c1.1 7.9 7.9 14 16 14h615.1c8 0 14.9-6 15.9-14l76.4-580.6c.8-6.7-6.5-11.4-12.3-7.9zm-126 534.1H250.3l-53.8-409.4 139.8 86.1L512 252.9l175.7 234.4 139.8-86.1-53.9 409.4zM512 509c-62.1 0-112.6 50.5-112.6 112.6S449.9 734.2 512 734.2s112.6-50.5 112.6-112.6S574.1 509 512 509zm0 160.9c-26.6 0-48.2-21.6-48.2-48.3 0-26.6 21.6-48.3 48.2-48.3s48.2 21.6 48.2 48.3c0 26.6-21.6 48.3-48.2 48.3z"></path></svg><!----></span></div>
            <div data-v-0f8075b3="" class="floating-icon" style="top: 75%; left: 25%; animation-delay: 2s;"><span data-v-0f8075b3="" role="img" aria-label="thunderbolt" class="anticon anticon-thunderbolt"><svg focusable="false" data-icon="thunderbolt" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M848 359.3H627.7L825.8 109c4.1-5.3.4-13-6.3-13H436c-2.8 0-5.5 1.5-6.9 4L170 547.5c-3.1 5.3.7 12 6.9 12h174.4l-89.4 357.6c-1.9 7.8 7.5 13.3 13.3 7.7L853.5 373c5.2-4.9 1.7-13.7-5.5-13.7zM378.2 732.5l60.3-241H281.1l189.6-327.4h224.6L487 427.4h211L378.2 732.5z"></path></svg><!----></span></div>
            <div data-v-0f8075b3="" class="floating-icon" style="top: 30%; right: 35%; animation-delay: 2.5s;"><span data-v-0f8075b3="" role="img" aria-label="heart" class="anticon anticon-heart"><svg focusable="false" data-icon="heart" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"></path></svg><!----></span></div>
          </div>
        </div>
      </div>
    </div>
    <div class="ant-card-body" style="display: flex; flex-direction: column; justify-content: center; padding: 16px 12px 0px 12px;">
      <div data-v-0f8075b3="" class="create-body">
        <div data-v-0f8075b3="" class="main-icon"><span data-v-0f8075b3="" role="img" aria-label="plus" class="anticon anticon-plus plus-icon"><svg focusable="false" data-icon="plus" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"></path><path d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"></path></svg><!----></span></div>
        <h3 data-v-0f8075b3="" class="create-title">Создать доску</h3>
        <p data-v-0f8075b3="" class="create-description">Создайте новую доску для вашего проекта</p>
        <div data-v-0f8075b3="" class="create-features">
          <div data-v-0f8075b3="" class="feature"><span data-v-0f8075b3="" role="img" aria-label="check" class="anticon anticon-check feature-icon"><svg focusable="false" data-icon="check" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 00-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"></path></svg><!----></span><span data-v-0f8075b3="">Шаблоны</span></div>
          <div data-v-0f8075b3="" class="feature"><span data-v-0f8075b3="" role="img" aria-label="check" class="anticon anticon-check feature-icon"><svg focusable="false" data-icon="check" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 00-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"></path></svg><!----></span><span data-v-0f8075b3="">Совместная работа</span></div>
        </div>
      </div>
    </div>
  </div>
</main>"
`;
